client/my-sites/hosting/staging-site-card/use-check-staging-site-status.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchAutomatedTransferStatus } from 'calypso/state/automated-transfer/actions';
+import { transferStates } from 'calypso/state/automated-transfer/constants';
+import {
+	getAutomatedTransferStatus,
+	isFetchingAutomatedTransferStatus,
+} from 'calypso/state/automated-transfer/selectors';
+
+export const useCheckStagingSiteStatus = ( siteId: number ) => {
+	const dispatch = useDispatch();
+
+	const transferStatus = useSelector( ( state ) => getAutomatedTransferStatus( state, siteId ) );
+	const isFetchingTransferStatus = useSelector( ( state ) =>
+		isFetchingAutomatedTransferStatus( state, siteId )
+	);
+
+	useEffect( () => {
+		if ( ! siteId || transferStatus === transferStates.COMPLETE ) {
+			return;
+		}
+
+		let transferInterval: NodeJS.Timeout;
+
+		if ( ! isFetchingTransferStatus ) {
+			transferInterval = setInterval(
```

Comment:
Initially, I tried to do it using `waitFor` like it's done in another component:

https://github.com/Automattic/wp-calypso/blob/f8823ab90e9a7b6dfa16c293abc32a8e35ca58aa/client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you.tsx#L118

Unfortunately, it was sending tens of requests to the server. Do you have any thoughts on what could cause it? Does `waitFor` also need a way to cancel the effect?

Reply:
<img width="1273" alt="Screenshot 2023-02-23 at 20 58 13" src="https://user-images.githubusercontent.com/779993/221031913-bfb817dc-8d8e-4be8-864b-bafe940082d0.png">

During my testint, the `setInterval` worked great. It waited 5-6 seconds between each request, and it only made 3 requests.